// directual-api/lib/api.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var endpoint_1 = require("./endpoint");
var auth_1 = require("./auth");
var defaultApiHost = 'https://api.directual.com';
var Config = /** @class */ (function () {
    function Config(config) {
        this.config = config;
    }
    return Config;
}());
exports.Config = Config;
var Structure = /** @class */ (function (_super) {
    __extends(Structure, _super);
    function Structure(name, config) {
        var _this = _super.call(this) || this;
        _this.name = name;
        _this.config = config;
        return _this;
    }
    return Structure;
}(endpoint_1.default));
exports.Structure = Structure;
var Api = /** @class */ (function () {
    function Api(config) {
        this.config = {};
        this.config = config;
        if (!config.apiHost) {
            this.config.apiHost = defaultApiHost;
        }
        this.auth = new auth_1.Auth(config);
    }
    Api.prototype.structure = function (name) {
        return new Structure(name, this.config);
    };
    return Api;
}());
exports.default = Api;
exports.__useDefault = true;
